tools/pipelines/templates/include-publish-npm-package.yml

```diff
@@ -32,6 +35,14 @@ stages:
       environment: test-package-build-feed
       pool: ${{ parameters.pool }}
 
+  # Only generate manifest files for runs in the internal project (i.e. CI runs, not PR run), for the main and release branch, and
+  # for the build of the client release group (we don't need manifests for anything else).
+  # Enabling this template for every PR run risks overwriting existing manifest files uploaded to Azure blobs. Therefore, it's crucial to restrict this template to commits merged into the main and release branch.
+  - ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(parameters.tagName, 'client'), or(eq(variables['Build.SourceBranch'], 'refs/heads/main'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/'))) }}:
+    - template: /tools/pipelines/templates/upload-dev-manifest.yml@self
+      parameters:
+        buildDirectory: '${{ parameters.buildDirectory }}'
```

Comment:
This stage only runs for test/ branches so the branch conditions here will never be true, I think we can just remove them in this case and keep them below.

Reply:
thanks for pointing this out. :)

Reply:
Sorry, I should have been clearer. The _branch_ conditions needed to go, but the rest I think we still need, particularly the tagName == client one; otherwise this would run for the other pipelines and probably fail.

Reply:
Unless we don't want the manifests at all for test/ branches... but I think they're necessary if we ever wanted to try integrating one of those into Loop?

Reply:
I guess we can avoid generating manifest files for test branches at this stage because packages are not being correctly pulled from the test feed, and there are versioning concerns. 

I'm planning is to merge the initial stage without including the test branches and only upload reports for packages published in the build feed. Additionally, the `isTestVersion` function should be removed and addressed in a follow-up task.

I have a [item](https://dev.azure.com/fluidframework/internal/_workitems/edit/8042) which captures work around test feed and publishing release reports for test feed.